pkg/controllers/resources/events/syncer.go

```diff
@@ -124,7 +111,9 @@ var _ syncer.ToVirtualSyncer = &eventSyncer{}
 
 func (s *eventSyncer) SyncToVirtual(ctx *synccontext.SyncContext, pObj client.Object) (ctrl.Result, error) {
 	// build the virtual event
-	vObj, err := s.buildVirtualEvent(ctx.Context, pObj.(*corev1.Event))
+	vObj := pObj.DeepCopyObject().(*corev1.Event)
+	vObj.ResourceVersion = ""
```

Comment:
This should be `translate.ResetObjectMetadata(vObj)`

Reply:
so if we don't call it manually beforehand, the resource version will still be set because vObj contains the resource version of the physical object at that point. 
https://github.com/loft-sh/vcluster/pull/1940/files#diff-841ade67d794007be038e9be601ecb81f2e9b7b2e5b10ec260d13345b54294c1R43 this is where we would be overwriting it. To avoid doing that we should have a parameter to know whether we are creating the object or not. I believe it's slightly worse than the current solution